merkle/tree_hasher.go

```diff
@@ -11,26 +11,33 @@ import (
 
 // Domain separation prefixes
 const (
-	LeafHashPrefix = 0
-	NodeHashPrefix = 1
+	RFC6962LeafHashPrefix = 0
```

Comment:
just to capure the IRL conversation: we agreed this needs to live somewhere else, but it's helpful since it calls out the fact that these things need to be configurable somehow.
Could you pop in some big scary comments and a TODO or two to sort all this out?